connectors/activemq/src/main/java/forklift/connectors/ActiveMQConnector.java

```diff
@@ -137,20 +139,45 @@ public ForkliftMessage jmsToForklift(Message m) {
     }
 
     @Override
-    public MessageProducer getQueueProducer(String name) {
-         try {
-            return getSession().createProducer(new ActiveMQQueue(name));
+    public ForkliftProducerI getQueueProducer(String name) {
+        try {
+            return new ActiveMQProducer(getSession().createProducer(new ActiveMQQueue(name)), getSession());
         } catch (JMSException | ConnectorException e) {
+            System.out.println("getQueueProducer, throwing error");
             return null;
         }
     }
-
+    
     @Override
-    public MessageProducer getTopicProducer(String name) {
+    public ForkliftProducerI getTopicProducer(String name) {
          try {
-            return getSession().createProducer(new ActiveMQTopic(name));
+            return new ActiveMQProducer(getSession().createProducer(new ActiveMQTopic(name)), getSession());
         } catch (JMSException | ConnectorException e) {
+            System.out.println("getTopicProducer, throwing error");
             return null;
         }
     }
+
+    @Override
+    public void register(Object instance) {
+        try {
+            Arrays.stream(instance.getClass().getDeclaredFields())
```

Comment:
Went ahead and refactored the injection out of the connector like we talked about, and kept it inside of the onMessage loop.